define the ietf-schedule module in a seperate draft, move it out from this draft (#28)
boucadair/policy-based-network-acl: draft-ma-netmod-schedule-yang.md

    city: Jiangsu
    code: 210012
    country: China
    email: maqiufang1@huawei.com
 -
    fullname: Qin Wu
    organization: Huawei
    street: 101 Software Avenue, Yuhua District
    city: Jiangsu
    code: 210012
    country: China
    email: bill.wu@huawei.com
 -
    fullname: Mohamed Boucadair
    organization: Orange
    city: Rennes
    code: 35000
    country: France
    email: mohamed.boucadair@orange.com
 -
    fullname: Daniel King
    organization: Lancaster University
    country: United Kingdom
    email: d.king@lancaster.ac.uk

normative:

informative:

--- abstract

   this document defines a common schedule YANG module which is
   designed to be applicable for scheduling information such as event, policy,
   services or resources based on date and time.

--- middle

# Introduction {#intro}

   This document defines a common schedule YANG module which
   is designed to be applicable for scheduling information such as event, policy,
   services or resources based on date and time. This model is designed with the intent to be reusable in other
   scheduling contexts. In particular, the groupings can be reused by other modules.

>    The YANG data models in this document conform to the Network

Reviewer: I would remove this mention as the module is about groupings and types.

#### Recurrence Rule

  The following snippet can be used to indicate a daily recurrent in December:

~~~~
{
  "recurrence": {
    "freq": "daily",
    "byyearmonth": [12]
  }
}
~~~~

   The following snippet can be used to indicate 10 occurrences that occur every last Saturday of the month:

~~~~
{
  "recurrence": {
    "freq": "monthly",
    "count": 10,
    "byday": [
      {
        "direction": [-1],
        "weekday": "saturday"
      }
    ]
  }
}
~~~~

   The following indicates the example of a recurrence that occurs on the last workday of the month until December 25, 2023:

~~~~
{
  "recurrence": {
    "freq": "monthly",
    "until": "2023-12-25",
    "byday": [
      { "weekday": "monday" },
      { "weekday": "tuesday" },
      { "weekday": "wednesday" },
      { "weekday": "thursday" },
      { "weekday": "friday" }
    ],
    "bysetpos": [-1]
  }
}
~~~~

   The following depicts the example of a recurrence that occurs every other week on Tuesday and Sunday, the week starts from Monday:

~~~
{
  "recurrence": {
    "freq": "weekly",
    "interval": 2,
    "byday": [
      { "weekday": "tuesday" },
      { "weekday": "sunday" }
    ],
    "wkst": "monday"
  }
}
~~~

> #  YANG Modules

Reviewer: ```suggestion

```

> ##  The "ietf-schedule" YANG Module {#sec-schedule}

Reviewer: ```suggestion
#  The "ietf-schedule" YANG Module {#sec-schedule}
```

~~~~

# Security Considerations

   The "ietf-schedule" YANG module specified in this document defines schema for data
   that is designed to be accessed via network management protocols such
   as NETCONF {{!RFC6241}} or RESTCONF {{!RFC8040}}.  The lowest NETCONF layer
   is the secure transport layer, and the mandatory-to-implement secure
   transport is Secure Shell (SSH) {{!RFC6242}}.  The lowest RESTCONF layer
   is HTTPS, and the mandatory-to-implement secure transport is TLS
   {{!RFC8446}}.

   The Network Configuration Access Control Model (NACM) {{!RFC8341}}
   provides the means to restrict access for particular NETCONF or
   RESTCONF users to a preconfigured subset of all available NETCONF or
   RESTCONF protocol operations and content.

   The "ietf-schedule" module defines a set of types and
   groupings.  These nodes are intended to be reused by other YANG
   modules.  The module by itself does not expose any data nodes that
   are writable, data nodes that contain read-only state, or RPCs.  As
   such, there are no additional security issues related to the "ietf-
   schedule" module that need to be considered.

#  IANA Considerations

##  The "IETF XML" Registry

>    This document registers the following URIs in the "IETF XML Registry" {{!RFC3688}}.

Reviewer: ```suggestion
   This document registers the following URI in the "IETF XML Registry" {{!RFC3688}}.
```

~~~~
        URI: urn:ietf:params:xml:ns:yang:ietf-schedule
        Registrant Contact: The IESG.
        XML: N/A, the requested URI is an XML namespace.
~~~~

##  The "YANG Module Names" Registry

>    This document registers the following YANG modules in the "YANG Module Names"

Reviewer: ```suggestion
   This document registers the following YANG module in the "YANG Module Names"
```

   registry {{!RFC6020}}.

~~~~
        name:               ietf-schedule
        namespace:          urn:ietf:params:xml:ns:yang:ietf-schedule
        prefix:             schedule
        maintained by IANA: N
        reference:          RFC XXXX
~~~~

--- back

# Changes between Revisions

# Acknowledgments
{:numbered="false"}

   This work is derived from the {{?I-D.ietf-opsawg-ucl-acl}}. There is a desire

>    from the WG to see this model be seperately defined for wide use in scheduling context.

Reviewer: ```suggestion
   from the OPSAWG to see this model be separately defined for wide use in scheduling context.
```

>    The authors would like to thank Adrian Farrel, Wei Pan, Tianran Zhou and Joe Clarke

Reviewer: ```suggestion
   The authors would like to thank Adrian Farrel, Wei Pan, Tianran Zhou and Joe Clarke
```
```suggestion
   Thanks to Adrian Farrel, Wei Pan, Tianran Zhou, and Joe Clarke
```

>    for their valuable comments and great input to this work.

Reviewer: ```suggestion
   for their valuable comments and inputs to this work.
```
